core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/action/AbstractDictionaryServlet.java

```diff
@@ -93,10 +105,38 @@ protected void service(@NotNull SlingHttpServletRequest request, @NotNull SlingH
         try {
             super.service(request, response);
         } catch (IllegalArgumentException e) {
-            LOG.error("Servlet parameter error: {}", e.getMessage());
+            LOG.error("{}: Servlet parameter error: {}", exceptionMessagePrefix, e.getMessage());
             HtmlResponse htmlResponse = new HtmlResponse();
             htmlResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
             htmlResponse.send(response, true);
         }
     }
+
+    @Override
+    protected void doPost(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response)
+            throws ServletException, IOException {
+        HtmlResponse htmlResponse = new HtmlResponse();
+        try {
+            internalDoPost(request, htmlResponse);
+            htmlResponse.send(response, true);
+        } catch (Throwable e) {
+            if (e instanceof IllegalArgumentException) {
+                throw (IllegalArgumentException)e; // rethrow IllegalArgumentException to be handled by the service method
+            }
+            LOG.error("{}: {}", exceptionMessagePrefix, e.getMessage(), e);
```

Comment:
stacktrace only with category DEBUG or leave it like this?

Reply:
for me it depends, if it's a client-caused error => debug, if it's an internal server error that we should look at => error

Reply:
that is not always unambigously to say, how would you classify a permission issue?
I for now would leave it at this level, later on we can relax if we get too reports (which I don't expect) but right now it helps a lot to improve the tool further to iron out the edge cases.

Reply:
The 404 cases should be already caught before.